File: ClientRuntimes/CSharp/ClientRuntime.Azure/Authentication/StringTokenProvider.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Threading;
using System.Threading.Tasks;

namespace Microsoft.Azure.Authentication
{
    /// <summary>
    /// A simple token provider that always provides a static access token.
    /// </summary>
    public sealed class StringTokenProvider : ITokenProvider
    {
        /// <summary>
        /// Create a token provider for bearer tokens that always returns the given access token.
        /// </summary>
        /// <param name="accessToken">The access token for this token provider to return.</param>
        public StringTokenProvider(string accessToken) : this(accessToken, "Bearer")
        {
        }

        /// <summary>
        /// Create a token provider for the given token type that returns the given access token.
        /// </summary>
        /// <param name="accessToken">The access token to return.</param>
        /// <param name="tokenType">The token type of the given access token.</param>
        public StringTokenProvider(string accessToken, string tokenType)
        {
            _accessToken = accessToken;
            _type = tokenType;
        }

        private string _accessToken;
        private string _type;

        /// <summary>
        /// Gets the token type of this access token.
        /// </summary>
        public string TokenType
        {
            get { return _type; }
        }

        /// <summary>
        /// Returns the access token.
        /// </summary>
        /// <param name="cancellationToken">The cancellation token for this action.  
        /// This will not be used since the returned token is static.</param>
        /// <returns>The access token.</returns>
        public Task<string> GetAccessTokenAsync(CancellationToken cancellationToken)
        {
            return Task.FromResult(_accessToken);
        }
    }
}

File: ClientRuntimes/Ruby/ms-rest/lib/ms_rest/credentials/string_token_provider.rb
# encoding: utf-8
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

module MsRest
  #
  # Class that provides access to authentication token.
  #
  class StringTokenProvider < TokenProvider

    private

    # @return [String] the access token.
    attr_accessor :token

    # @return [String] the token type.
    attr_accessor :token_type

    public

    #
    # Creates and initalizes a new instance of StringTokenProvider class.

    # @param token [String] the access token.
    # @param token_type [String] the token type.
    #
    def initialize(token, token_type = TokenCredentials::DEFAULT_SCHEME)
      @token = token
      @token_type = token_type
    end

    #
    # Returns the string value which needs to be attached
    # to HTTP request header in order to be authorized.
    #
    # @return [String] authentication headers.
    def get_authentication_header
      "#{token_type} #{token}"
    end
  end
end
